sdk/storage/Azure.Storage.Blobs/src/LeaseClient.cs

```diff
@@ -0,0 +1,1091 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Common;
+
+namespace Azure.Storage.Blobs.Specialized
+{
+    /// <summary>
+    /// The <see cref="LeaseClient"/> allows you to manipulate Azure
+    /// Storage leases on containers and blobs.
+    /// </summary>
+	public class LeaseClient
+    {
+        /// <summary>
+        /// The <see cref="BlobClient"/> to manage leases for.
+        /// </summary>
+        private readonly BlobClient _blob;
+
+        /// <summary>
+        /// The <see cref="BlobContainerClient"/> to manage leases for.
+        /// </summary>
+        private readonly BlobContainerClient _container;
+
+        /// <summary>
+        /// Gets the URI of the object being leased.
+        /// </summary>
+        public Uri Uri => this._blob?.Uri ?? this._container?.Uri;
+
+        /// <summary>
+        /// Gets the Lease ID for this lease.
+        /// </summary>
+        public string LeaseId { get; private set; }
+
+        /// <summary>
+        /// The <see cref="HttpPipeline"/> transport pipeline used to send
+        /// every request.
+        /// </summary>
+        private HttpPipeline Pipeline => this._blob?.Pipeline ?? this._container._pipeline;
```

Comment:
They can pass it to the Blob or Container client that they use to get a LeaseClient.  And yes, we don't have all 7 ctors for LeaseClient, but I thought per the review discussion we were okay with that since it's tied to a given blob or container.